extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/videoQualityPanel.js

```diff
@@ -0,0 +1,109 @@
+    };
+  },
+
+  handleVideoQualityClick: function(selectedBitrateId, event) {
+    event.preventDefault();
+    var eventData = {
+      "id": selectedBitrateId
+    };
+    this.props.controller.sendVideoQualityChangeEvent(eventData);
+    this.setState({
+      selected: selectedBitrateId
+    });
+    this.props.togglePopoverAction();
+  },
+
+  render: function() {
+    var availableBitrates  = this.props.videoQualityOptions.availableBitrates;
+    var bitrateButtons = [];
+    var label;
+
+    //available bitrates
+    for (var i = 0; i < availableBitrates.length; i++) {
+      var qualityBtn = ClassNames({
+        'oo-quality-btn': true,
+        'oo-selected': this.state.selected == availableBitrates[i].id
+      });
+      var selectedBitrateStyle = {color: (this.props.skinConfig.general.accentColor && this.state.selected == availableBitrates[i].id) ? this.props.skinConfig.general.accentColor : null};
+
+      if (availableBitrates[i].id === 'auto') {
+        label = 'Auto';
+      } else if (typeof availableBitrates[i].height === "number") {
+				label = Math.round(availableBitrates[i].height) + 'p';
+			}
+			else {
+				label = availableBitrates[i].bitrate;
+			}
+			bitrateButtons.push(<li key={i}><a className={qualityBtn} style={selectedBitrateStyle} key={i} onClick={this.handleVideoQualityClick.bind(this, availableBitrates[i].id)}>{label}</a></li>);
+    }
+
+    var qualityScreenClass = ClassNames({
+      'oo-content-panel': !this.props.popover,
+      'oo-quality-panel': !this.props.popover,
+      'oo-quality-popover': this.props.popover,
+      'oo-mobile-fullscreen': !this.props.popover && this.props.controller.state.isMobile && (this.props.controller.state.fullscreen || this.props.controller.state.isFullWindow)
+    });
+
+    return (
+      <div className={qualityScreenClass}>
+        <ScrollArea
+          className="oo-quality-screen-content"
+          speed={this.props.popover ? 0.6 : 1}
+          horizontal={!this.props.popover}>
+          <ul>
+            {bitrateButtons}
+          </ul>
+        </ScrollArea>
+      </div>
+    );
+  }
+});
+
+VideoQualityPanel.propTypes = {
+  videoQualityOptions: React.PropTypes.shape({
+    availableBitrates: React.PropTypes.arrayOf(React.PropTypes.shape({
+      id: React.PropTypes.string,
+      bitrate: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.number,
+      ]),
+      label: React.PropTypes.string
+    }))
+  }),
+  togglePopoverAction: React.PropTypes.func,
+  controller: React.PropTypes.shape({
+    sendVideoQualityChangeEvent: React.PropTypes.func
+  })
+};
+
+VideoQualityPanel.defaultProps = {
+  popover: false,
+  skinConfig: {
+    icons: {
+      quality:{fontStyleClass:'oo-icon oo-icon-topmenu-quality'}
+    }
+  },
+  videoQualityOptions: {
+    availableBitrates: []
+  },
+  togglePopoverAction: function(){},
+  controller: {
+    sendVideoQualityChangeEvent: function(a){}
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "a". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)